folium/features.py

```diff
@@ -1095,6 +1095,9 @@ def get_by_key(obj, key):
 
             def color_scale_fun(x):
                 key_of_x = get_by_key(x, key_on)
+                if not key_of_x:
+                    if len(key_on) > 0:
```

Comment:
In what case will `key_on` be an empty string? And why don't we want to raise an error in that case? If `key_on` is indeed an empty string it will still fail, so good to raise an error I would say.